contracts/access/manager/AccessManager.sol

```diff
@@ -0,0 +1,703 @@
+        if (!hasGroup(groupId, msgsender)) {
+            revert AccessControlUnauthorizedAccount(msgsender, groupId);
+        }
+        _;
+    }
+
+    /**
+     * @dev Check that the caller is an admin and that the top-level function currently executing has been scheduled
+     * sufficiently ahead of time, if necessary according to admin delays.
+     */
+    modifier onlyDelayedAdmin() {
+        address msgsender = _msgSender();
+        (bool allowed, uint32 delay) = canCall(msgsender, address(this), msg.sig);
+        if (delay > 0) {
+            _consumeScheduledOp(_hashOperation(_msgSender(), address(this), _msgData()));
+        } else if (!allowed) {
+            revert AccessControlUnauthorizedAccount(msgsender, ADMIN_GROUP);
+        }
+        _;
+    }
+
+    constructor(address initialAdmin) {
+        // admin is active immediately and without any execution delay.
+        _grantGroup(ADMIN_GROUP, initialAdmin, 0, 0);
+    }
+
+    // =================================================== GETTERS ====================================================
+    /**
+     * @dev Check if an address (`caller`) is authorised to call a given function on a given contract directly (with
+     * no restriction). Additionally, it returns the delay needed to perform the call indirectly through the {schedule}
+     * & {relay} workflow.
+     *
+     * This function is usually called by the targeted contract to control immediate execution of restricted functions.
+     * Therefore we only return true is the call can be performed without any delay. If the call is subject to a delay,
+     * then the function should return false, and the caller should schedule the operation for future execution.
+     *
+     * We may be able to hash the operation, and check if the call was scheduled, but we would not be able to cleanup
+     * the schedule, leaving the possibility of multiple executions. Maybe this function should not be view?
+     *
+     * NOTE: The IAuthority interface does not include the `uint32` delay. This is an extension of that interface that
+     * is backward compatible. Some contract may thus ignore the second return argument. In that case they will fail
+     * to identify the indirect workflow, and will consider call that require a delay to be forbidden.
+     */
+    function canCall(address caller, address target, bytes4 selector) public view virtual returns (bool, uint32) {
+        AccessMode mode = getContractMode(target);
+        if (mode == AccessMode.Open) {
+            return (true, 0);
+        } else if (mode == AccessMode.Closed) {
+            return (false, 0);
+        } else if (caller == address(this)) {
+            // Caller is AccessManager => call was relayed. In that case the relay already checked permissions. We
+            // verify that the call "identifier", which is set during the relay call, is correct.
+            return (_relayIdentifier == keccak256(abi.encodePacked(target, selector)), 0);
```

Comment:
I'd suggest putting `keccak256(abi.encodePacked(target, selector))` in a private function like:

```solidity
function relayIdentifier(address target, bytes4 selector) private pure returns (bytes32) {
  return keccak256(abi.encodePacked(target, selector));
}
```

The reason is that I think it's better if the definition of the identifier is encapsulated.